### PR TITLE
Test case to validate Multipart Upload on buckets

### PR DESCRIPTION
**Pull request by parikshithb**
This test case covers Multipart Upload which allows the user to upload a single object as a set of parts. After all parts of your object are uploaded, S3 then presents the data as a single object. With this feature you can create parallel uploads.

### CODE REVIEW

File: ocs_ci/ocs/resources/mcg.py [hunk 1]
```diff
@@ -618,3 +624,86 @@ def _check_state():
...
+
+        Returns:
+            dict : Dictionary containing the completed multipart upload details
+
+        """
+        result = self.s3_client.complete_multipart_upload(
+            Bucket=bucketname,
+            Key=object_key,
+            UploadId=upload_id,
+            MultipartUpload={"Parts": parts})
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/mcg.py]: ) in new line

File: ocs_ci/ocs/resources/mcg.py [hunk 2]
```diff
@@ -618,3 +624,86 @@ def _check_state():
...
+        """
+        aborted = []
+        multipart_list = self.s3_client.list_multipart_uploads(Bucket=bucketname)
+        print("Aborting", len(multipart_list), "uploads")
+        if "Uploads" in multipart_list:
+            for i in multipart_list["Uploads"]:
+                upload_id = i["UploadId"]
+                aborted.append(
+                    self.s3_client.abort_multipart_upload(
+                        Bucket=bucketname, Key=object_key, UploadId=upload_id))
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/mcg.py]: ) in new line

File: ocs_ci/ocs/resources/mcg.py [hunk 3]
```diff
@@ -618,3 +624,86 @@ def _check_state():
...
+            bucketname (str): Name of the bucket
+            object_key (str): Unique object Identifier
+
+        Returns:
+            list : List of aborted upload ids
+
+        """
+        aborted = []
+        multipart_list = self.s3_client.list_multipart_uploads(Bucket=bucketname)
+        print("Aborting", len(multipart_list), "uploads")
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/mcg.py]: please use log.info instead of print

File: tests/manage/mcg/helpers.py [hunk 4]
```diff
@@ -118,3 +118,33 @@ def write_individual_s3_objects(mcg_obj, awscli_pod, bucket_factory, downloaded_
...
+        uploaded_parts (list): list containing the name of the parts to be uploaded
+
+    Returns:
+        list: List containing the ETag of the parts
+
+    """
+    parts = []
+    secrets = [mcg_obj.access_key_id, mcg_obj.access_key, mcg_obj.s3_endpoint]
+    for count, part in enumerate(uploaded_parts, 1):
+        upload_cmd = f'upload-part --bucket {bucketname} --key {object_key}' \
```
Comment (RazTamir) [inline tests/manage/mcg/helpers.py]:
```
upload_cmd = (
    f'upload-part --bucket {bucketname} --key {object_key}'
    f' --part-number {count} --body {body_path}/{part} '
    f'--upload-id {upload_id}'
)
```

File: tests/manage/mcg/helpers.py [hunk 5]
```diff
@@ -118,3 +118,33 @@ def write_individual_s3_objects(mcg_obj, awscli_pod, bucket_factory, downloaded_
...
+            f' --part-number {count} --body {body_path}/{part} ' \
+            f'--upload-id {upload_id}'
+        part = awscli_pod.exec_cmd_on_pod(
+            command=craft_s3_api_command(mcg_obj, upload_cmd), out_yaml_format=False,
+            secrets=secrets
+        ).split("\"")[-3].split("\\")[0]
```
Comment (RazTamir) [inline tests/manage/mcg/helpers.py]: please add a comment on what is happening here

File: tests/manage/mcg/test_multipart_upload.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        logger.info(f'Aborting any Multipart Upload on bucket:{bucket}')
+        mcg_obj.abort_multipart_upload(bucket, key)
+
+        # Create & list Multipart Upload on the Bucket
+        logger.info(f'Initiating Multipart Upload on Bucket: {bucket} with Key {key}')
+        upload_id = mcg_obj.create_multipart_upload(bucket, key)
+        logger.info(f'Listing the Multipart Upload : {mcg_obj.list_multipart_upload(bucket)}')
+
+        # Uploading individual parts to the Bucket
+        uploaded_parts = helpers.upload_parts(mcg_obj, awscli_pod, bucket, key, res_dir, upload_id, parts)
```
Comment (RazTamir) [inline tests/manage/mcg/test_multipart_upload.py]: add looger info

File: ocs_ci/ocs/resources/mcg.py [hunk 7]
```diff
@@ -618,3 +624,86 @@ def _check_state():
...
+        if "Uploads" in multipart_list:
+            for i in multipart_list["Uploads"]:
+                upload_id = i["UploadId"]
+                aborted.append(
+                    self.s3_client.abort_multipart_upload(
+                        Bucket=bucketname, Key=object_key, UploadId=upload_id))
+        return aborted
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py]:
```suggestion
    multipart_list = self.s3_client.list_multipart_uploads(Bucket=bucketname)
    log.info(f"Aborting{len(multipart_list)} uploads")
    if "Uploads" in multipart_list:
        return [
            self.s3_client.abort_multipart_upload(
                Bucket=bucketname, Key=object_key, UploadId=upload["UploadId"]
            ) for upload in multipart_list["Uploads"]
        ]
    else:
        return None
```

File: tests/manage/mcg/helpers.py [hunk 8]
```diff
@@ -118,3 +118,33 @@ def write_individual_s3_objects(mcg_obj, awscli_pod, bucket_factory, downloaded_
             command=craft_s3_command(mcg_obj, copycommand), out_yaml_format=False,
             secrets=[mcg_obj.access_key_id, mcg_obj.access_key, mcg_obj.s3_endpoint]
         )
+
+
+def upload_parts(mcg_obj, awscli_pod, bucketname, object_key, body_path, upload_id, uploaded_parts):
+    """
+    Uploads individual parts to a bucket
```
Comment (ebenahar) [inline tests/manage/mcg/helpers.py]: add an empty line below

File: tests/manage/mcg/helpers.py [hunk 9]
```diff
@@ -118,3 +118,33 @@ def write_individual_s3_objects(mcg_obj, awscli_pod, bucket_factory, downloaded_
...
+    Args:
+        mcg_obj (obj): An MCG object containing the MCG S3 connection credentials
+        awscli_pod (pod): A pod running the AWSCLI tools
+        bucketname: Name of the bucket to upload parts on
```
Comment (ebenahar) [inline tests/manage/mcg/helpers.py]: bucketname (str):

File: ocs_ci/ocs/resources/mcg.py (hunk 1 above)
Comment (parikshithb) [reply ocs_ci/ocs/resources/mcg.py]: ack

File: ocs_ci/ocs/resources/mcg.py (hunk 2 above)
Comment (parikshithb) [reply ocs_ci/ocs/resources/mcg.py]: ack

File: ocs_ci/ocs/resources/mcg.py (hunk 3 above)
Comment (parikshithb) [reply ocs_ci/ocs/resources/mcg.py]: ack

File: tests/manage/mcg/helpers.py (hunk 4 above)
Comment (parikshithb) [reply tests/manage/mcg/helpers.py]: ack

File: tests/manage/mcg/helpers.py (hunk 5 above)
Comment (parikshithb) [reply tests/manage/mcg/helpers.py]: Ack

File: tests/manage/mcg/test_multipart_upload.py (hunk 6 above)
Comment (parikshithb) [reply tests/manage/mcg/test_multipart_upload.py]: Added

File: ocs_ci/ocs/resources/mcg.py (hunk 7 above)
Comment (parikshithb) [reply ocs_ci/ocs/resources/mcg.py]: Thanks for the changes :)

File: tests/manage/mcg/helpers.py (hunk 8 above)
Comment (parikshithb) [reply tests/manage/mcg/helpers.py]: ack

File: tests/manage/mcg/helpers.py (hunk 9 above)
Comment (parikshithb) [reply tests/manage/mcg/helpers.py]: ack

File: ocs_ci/ocs/resources/mcg.py [hunk 10]
```diff
@@ -697,13 +698,13 @@ def abort_multipart_upload(self, bucketname, object_key):
...
-                aborted.append(
-                    self.s3_client.abort_multipart_upload(
-                        Bucket=bucketname, Key=object_key, UploadId=upload_id))
-        return aborted
+            return [
+                self.s3_client.abort_multipart_upload(
+                    Bucket=bucketname, Key=object_key, UploadId=upload["UploadId"]
+                ) for upload in multipart_list["Uploads"]
+            ]
+        else:
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/mcg.py]: not a big deal but no need for the 'else' part as it will just return None in case the 'if' above is False

**Pull request merged**